lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -1938,17 +1946,28 @@ public function NewObjectExpression()
     }
 
     /**
-     * NewObjectArg ::= ScalarExpression | "(" Subselect ")"
+     * NewObjectArg ::= ScalarExpression | NamedScalarExpression | "(" Subselect ")"
      *
      * @return mixed
      */
-    public function NewObjectArg()
+    public function NewObjectArg(bool $namedArgAlreadyParsed = false)
```

Comment:
The Parser is neither final nor internal. This means this is a breaking change for extending classes. You can obtain the new argument with `func_get_args()` without changing the signature.
```suggestion
    public function NewObjectArg(/* bool $namedArgAlreadyParsed = false */)
```